### PR TITLE
Bp govuk sign in2

### PR DESCRIPTION
**Pull request by BarryPiccinni**
Add the \account\sign-in endpoint in cica-web

### CODE REVIEW

File: account/routes.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const express = require('express');
+const signInService = require('../govuk/sign-in/index')();
```
Comment (armoj) [inline account/routes.js:4]: move the "creation" of the service in to the route itself

**Review by armoj — CHANGES_REQUESTED** [review]

File: account/routes.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const router = express.Router();
+
+router.get('/sign-in', async (req, res, next) => {
+    try {
+        const redirectUri = `${req.protocol}://${req.get('host')}/account/signed-in`;
+        const url = await signInService.getServiceUrl(redirectUri);
+        return res.redirect(302, url);
+    } catch (err) {
+        res.status(err.statusCode || 404).render('404.njk');
```
[the remark this answers is not in the document]
Comment (BarryPiccinni) [reply account/routes.js:14]: I'm going to leave this at the moment. We don't have a suitable 503 page to cover this, I will ask Graeme to prepare some content for an error page which reflect this scenario.

File: govuk/sign-in/authorisation/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+function createAuthorisationService() {
+    function getAuthorisationURI(issuer, redirectUri) {
+        const opts = {
+            scope: ['openid'],
+            response_type: 'code',
+            client_id: 'RSsuV9e2KzU9IgtvOJsyZIOVy8U',
+            redirect_uri: redirectUri,
+            state: 'STATE',
+            nonce: 'NONCE',
+            vtr: '[Cl.Cm]'
```
[the remark this answers is not in the document]
Comment (BarryPiccinni) [reply govuk/sign-in/authorisation/index.js:12]: Yes. This does.

File: govuk/sign-in/authorisation/index.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+function createAuthorisationService() {
+    function getAuthorisationURI(issuer, redirectUri) {
+        const opts = {
+            scope: ['openid'],
```
[the remark this answers is not in the document]
Comment (BarryPiccinni) [reply govuk/sign-in/authorisation/index.js:6]: Yep, the array is unnecessary.

File: govuk/sign-in/authorisation/index.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            response_type: 'code',
+            client_id: 'RSsuV9e2KzU9IgtvOJsyZIOVy8U',
+            redirect_uri: redirectUri,
+            state: 'STATE',
```
[the remark this answers is not in the document]
Comment (BarryPiccinni) [reply govuk/sign-in/authorisation/index.js]: I have implemented a set of default values with the option to overwrite these by passing them in. Thoughts welcome!

**Review by armoj — APPROVED** [review]

**Pull request merged**